I'd love to make a patch for NativeAot.
Currently, I'm having a trouble with setting up NativeAot test env like below.
```
dlerror.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.linux-x64 with version (= 7.0.0-preview.5.22301.12)
dlerror.csproj : error NU1102:   - Found 1 version(s) in local [ Nearest version: 7.0.0-dev ]
dlerror.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-x64. No packages exist with this id in source(s): local
```

This is my nuget.config.
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="local" value="${runtime}/artifacts/packages/Release/Shipping" />
    <!--add key="nuget" value="https://api.nuget.org/v3/index.json" /-->
  </packageSources>
</configuration>
```

Please help me setting up the test env.

--- a/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs
+++ b/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs
@@ -51,7 +51,11 @@ namespace System.Runtime.InteropServices
 
             public void TrackErrorMessage(string? message)
             {
-                _errorMessage = message;
+                if (_errorMessage == null)
+                {
+                    _errorMessage = Environment.NewLine;
+                }
+                _errorMessage += " " + message + Environment.NewLine;
             }
         }
     }

--- a/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs
+++ b/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs
@@ -55,7 +55,10 @@ namespace System.Runtime.InteropServices
                 {
                     _errorMessage = Environment.NewLine;
                 }
-                _errorMessage += " " + message + Environment.NewLine;
+                if (!_errorMessage.Contains(message))
+                {
+                    _errorMessage += message + Environment.NewLine;
+                }
             }
         }
     }

--- a/src/coreclr/vm/nativelibrary.cpp
+++ b/src/coreclr/vm/nativelibrary.cpp
@@ -53,7 +53,7 @@ namespace
             LIMITED_METHOD_CONTRACT;
             m_hr = E_FAIL;
             m_priorityOfLastError = 0;
-            m_message = SString(SString::Utf8, "");
+            m_message = SString(SString::Utf8, "\n");
         }
 
         VOID TrackErrorCode()
@@ -139,14 +139,14 @@ namespace
         void SetMessage(LPCSTR message)
         {
 #ifdef TARGET_UNIX
-            //Append dlerror() messages of all attempts
+            //Append dlerror() messages
             SString new_message = SString(SString::Utf8, message);
             SString::Iterator i = m_message.Begin();
-
             if (!m_message.Find(i, new_message))
             {
-                m_message += new_message;
                 m_message += SString(SString::Utf8, " ");
+                m_message += new_message;
+                m_message += SString(SString::Utf8, "\n");
             }
 #else
             m_message = SString(SString::Utf8, message);

--- a/src/coreclr/vm/nativelibrary.cpp
+++ b/src/coreclr/vm/nativelibrary.cpp
@@ -144,7 +144,6 @@ namespace
             SString::Iterator i = m_message.Begin();
             if (!m_message.Find(i, new_message))
             {
-                m_message += SString(SString::Utf8, " ");
                 m_message += new_message;
                 m_message += SString(SString::Utf8, "\n");
             }

